impls/src/client_utils/client.rs

```diff
@@ -165,7 +165,7 @@ impl Client {
 	/// provided JSON object as body on a given URL that returns a future. Handles
 	/// request building, JSON serialization and deserialization, and response code
 	/// checking.
-	pub fn _post_async<IN, OUT>(
+	pub fn post_async<IN, OUT>(
```

Comment:
Out of curiosity why not using `pub async fn`?

Reply:
This code is originally from the [grin repo](https://github.com/mimblewimble/grin/blob/6e5afe496b201a1a5df4f42c62cd5fe1bdc24c6b/api/src/client.rs).
[My PR there](https://github.com/mimblewimble/grin/pull/3214/files#diff-1ba0421725cb0e5e93523cb69ded4a46) is updating the code to use `pub async fn`, although iirc the wallet only uses the blocking function. Using the blocking functions many times is not really optimal since it spawns a full runtime for each request, but its something we can improve in a separate PR. For this PR i think its fine to leave as is

Reply:
Indeed, focus is elsewhere on this PR, can likely make improvements later.